CMakeLists.txt

```diff
@@ -1,10 +1,65 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.0)
 project(termcolor)
 
-if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  set(CMAKE_CXX_FLAGS "-Werror -Wall -pedantic")
+# define target to install and link tests against
+add_library(termcolor INTERFACE)
+target_include_directories(termcolor
+  INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+
+# ALIAS same as in configure file
+# the ALIAS provides the same target when using `add_subdirectory()`
+# and when using `find_package(termcolor CONFIG)`. In both cases
+# the target `termcolor::termcolor` can be used.
+# create the alias termcolor::termcolor
+add_library(termcolor::termcolor ALIAS termcolor)
+
+# add compiler flags for test targets only for GCC and Clang
+target_compile_options(${PROJECT_NAME} INTERFACE
```

Comment:
good catch!
Because of the `BUILD_INTERFACE` part in the definition these flags would not propagate to the installed target. But when used with `add_subdirectory` these flags would propagate. Will fix